robot/impl/robot_reconfigure_weak_dependencies_test.go

```diff
@@ -285,4 +292,365 @@ func TestUpdateWeakDependents(t *testing.T) {
 	test.That(t, weak1.resources, test.ShouldHaveLength, 2)
 	test.That(t, weak1.resources, test.ShouldContainKey, base1Name)
 	test.That(t, weak1.resources, test.ShouldContainKey, base2Name)
+	test.That(t, weak1.reconfigCount, test.ShouldEqual, 3)
+}
+
+func TestWeakDependentsExplicitDependency(t *testing.T) {
```

Comment:
Awesome new test here and below; thanks for taking the time here. I myself have put up too many PRs around this stuff without a test.